subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c

```diff
@@ -162,37 +181,47 @@ int nrf_cloud_coap_pgps_url_get(struct nrf_cloud_rest_pgps_request const *const
 	size_t len = sizeof(buffer);
 	int err;
 
+	/* Take the semaphore before modifying the static buffer */
+	(void)k_sem_take(&coap_transfer_sem, K_FOREVER);
+
 	err = coap_codec_pgps_encode(request, buffer, &len,
 				     COAP_CONTENT_FORMAT_APP_CBOR);
 	if (err) {
 		LOG_ERR("Unable to encode P-GPS request: %d", err);
+		k_sem_give(&coap_transfer_sem);
 		return err;
 	}
 
+	pgps_err = 0;
+
 	err = nrf_cloud_coap_fetch(COAP_PGPS_RSC, NULL,
 				   buffer, len, COAP_CONTENT_FORMAT_APP_CBOR,
 				   COAP_CONTENT_FORMAT_APP_CBOR, true, get_pgps_callback,
 				   file_location);
+
 	if (!err && !pgps_err) {
-		LOG_INF("Got P-GPS file location");
-	} else if (err == -EAGAIN) {
-		LOG_ERR("Timeout waiting for P-GPS file location");
+		LOG_DBG("Got P-GPS URL");
+	} else if (err) {
+		if (err == -EAGAIN) {
+			LOG_ERR("Timeout waiting for P-GPS URL");
+		} else {
+			LOG_ERR("CoAP client error: %d", err);
+		}
 	} else {
-		if (pgps_err > 0) {
-			LOG_RESULT_CODE_ERR("Error getting P-GPS file location; result code:",
-					    pgps_err);
-			err = pgps_err;
+		err = pgps_err;
+		if (err > 0) {
+			LOG_RESULT_CODE_ERR("Error getting P-GPS URL; result code:", err);
 		} else {
-			err = pgps_err;
-			LOG_ERR("Error: %d", err);
+			LOG_ERR("P-GPS response decoding error: %d", err);
 		}
 	}
 
+	k_sem_give(&coap_transfer_sem);
```

Comment:
i don't have a huge preferences... i changed the original format because it was hard for me to follow.

the new format is basically: check for success, deal with the `err` value, deal with the coap error value.
so yes, the `else` is implicitly handling the coap error value.

if your handling format is preferred i can do that.

Reply:
I definitely prefer my suggested way, ***but*** IDK that I prefer it enough to force you a different direction, so I guess let's just let @plskeggs be the tie breaker?